gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java

```diff
@@ -50,6 +52,12 @@
   // Used for single-byte reads.
   private final byte[] singleReadBuf = new byte[1];
 
+  // Statistics tracker of Input Stream
```

Comment:
Can we move all the instrumentation code to a separate class, and use it as a delegate ?

Reply:
@medb @mayanks Thoughts on extracting instrumentation code into a sub-class/ proxy-decorator layer ?

Reply:
agree with @mprashanthsagar comment. It will keep this class (and code) clean with respect to inputStream interface and leave the stats handling to the delegate.

Reply:
We will work on it